geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/vlm/RasterSource.scala

```diff
@@ -76,24 +85,121 @@ object RasterSource {
     ProjectedRasterLayer(projectedRasterRDD)
   }
 
+  def readOrdered(
+    sc: SparkContext,
+    layerType: String,
+    paths: java.util.ArrayList[java.util.HashMap[String, String]],
+    targetCRS: String,
+    numPartitions: Integer,
+    resampleMethod: ResampleMethod,
+    readMethod: String
+  ): ProjectedRasterLayer = {
+    val scalaPaths: Seq[Seq[(String, String)]] = paths.asScala.toSeq.map { _.asScala.toSeq }
+
+    val partitions =
+      numPartitions match {
+        case i: Integer => Some(i.toInt)
+        case null => None
+      }
+
+    readOrdered(
+      sc,
+      layerType,
+      sc.parallelize(scalaPaths, partitions.getOrElse(sc.defaultParallelism)),
+      targetCRS,
+      resampleMethod,
+      readMethod
+    )
+  }
+
+  def readOrdered(
+    sc: SparkContext,
+    layerType: String,
+    rdd: RDD[Seq[(String, String)]],
+    targetCRS: String,
+    resampleMethod: ResampleMethod,
+    readMethod: String
+  ): ProjectedRasterLayer = {
+    val rasterSourcesRDD: RDD[Seq[(String, RasterSource)]] =
+      (readMethod match {
+        case GEOTRELLIS =>
+          rdd.mapPartitions { iter =>
+            iter.map { files: Seq[(String, String)] =>
+              files.map { case (k, v) => (k, new GeoTiffRasterSource(v): RasterSource)
+              }
+            }
+          }
+        case GDAL =>
+          rdd.mapPartitions { iter =>
+            iter.map { files: Seq[(String, String)] =>
+              files.map { case (k, v) => (k, GDALRasterSource(v): RasterSource)
+              }
+            }
+          }
+      }).cache()
+
+    val orderedRasterSourcesRDD: RDD[Seq[RasterSource]] =
+      rasterSourcesRDD.map { keyedSource: Seq[(String, RasterSource)] =>
+        keyedSource.sortBy { _._1 }.map { case (_, source) => source }
```

Comment:
sort here is not sufficient. Just due to shape of the input it may be possible to have data for bands "2" and "3" but not "1" for a given location.

Reply:
There needs to be a hard mapping from source to target bands such that `NODATA` tiles can be generated in above cases.